packages/modular-scripts/src/lint.ts

```diff
@@ -15,121 +14,128 @@ import {
   partitionPackages,
 } from './utils/unobtrusiveModular';
 export interface LintOptions {
-  all: boolean;
   fix: boolean;
   staged: boolean;
-  packages?: string[];
+  regex?: string[];
   ancestors?: boolean;
   descendants?: boolean;
   changed?: boolean;
-  includeNonModular?: boolean;
   compareBranch?: string;
+  diff?: boolean;
 }
 
 async function lint(
   options: LintOptions,
-  userRegexes: string[] = [],
+  packages: string[] = [],
 ): Promise<void> {
   const {
-    all = false,
     fix = false,
     staged = false,
-    packages = [],
+    regex: userRegexes = [],
     ancestors = false,
     descendants = false,
     changed = false,
-    includeNonModular = false,
     compareBranch,
+    diff = false,
   } = options;
   const modularRoot = getModularRoot();
   const lintExtensions = ['.ts', '.tsx', '.js', '.jsx'];
 
   const [workspaceMap] = await getAllWorkspaces();
-  const isSelective =
-    packages?.length || ancestors || descendants || changed || compareBranch;
 
-  let selectedTargets: string[];
+  const isSelective = changed || ancestors || descendants || packages.length;
 
-  if (!isSelective) {
-    selectedTargets = [...workspaceMap.keys()];
-  } else {
-    // Otherwise, calculate which packages are selected
-    selectedTargets = await selectWorkspaces({
-      targets: packages,
-      changed,
-      ancestors,
-      descendants,
-      compareBranch,
-    });
-  }
-
-  // Split packages into modular and non-modular testable. Make sure that "root" is not there.
-  const [modularTargets, nonModularTargets] = partitionPackages(
-    selectedTargets,
-    workspaceMap,
-    'lint',
-  );
-
-  // Compute patterns to pass Jest for the packages that we want to test
-  const selectedPackageRegexes = await computeRegexesFromPackageNames(
-    modularTargets,
-  );
-
-  // Only set regexes if all or selective options have been specified
-  let processedPackageRegexes = selectedPackageRegexes.map(
-    (regex) => `<rootDir>/${regex}/src/**/*.{js,jsx,ts,tsx}`,
-  );
-
-  let lintRegexes = isSelective || all ? processedPackageRegexes : [];
-
-  if (all && isSelective) {
-    logger.warn(
-      'You specified --all already; selective options are redundant.',
+  if ((staged || diff || userRegexes.length) && isSelective) {
+    logger.error(
+      "Can't specify --staged --diff --regex along with selective options",
```

Comment:
Slightly more explicit:

```suggestion
      "Can't specify --staged, --diff or --regex along with selective options",
```